validator/std_validator.go

```diff
@@ -0,0 +1,70 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// Validate validates the cobra commands
+// and all the descendants
+func Validate(cmd *cobra.Command) error {
+	err := handleDefaultErrors(cmd)
+	if err != nil {
+		return err
+	}
+
+	// handeling errors for subcommands of a command
+	subCommmands := cmd.Commands()
+	for _, child := range subCommmands {
+		err := handleDefaultErrors(child)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// ValidateCustom will allow user to
+// write their own custom rules for validation
+// over the default rules
+func ValidateCustom(cmd *cobra.Command, handleErrors func() error) error {
+	err1 := Validate(cmd)
+	if err1 != nil {
+		return err1
+	}
+	err2 := handleErrors()
+	if err2 != nil {
+		return err2
+	}
+
+	return nil
+}
+
+// handleDefaultErrors handles the default errors
+func handleDefaultErrors(cmd *cobra.Command) error {
+	cmdPath := cmd.CommandPath()
+
+	if len(cmd.Use) < 1 {
```

Comment:
Makes no sense in number of dimensions:


- we are missing exact location that can be used to pinpoint file that causing issues
- we are returning only single issue (rather than numerous issues during linting)
- Hardcoded values.